xoai-data-provider/src/main/java/io/gdcc/xoai/dataprovider/handlers/ListIdentifiersHandler.java

```diff
@@ -69,8 +70,10 @@ else if (parameters.hasFrom() && parameters.hasUntil())
                 listItemIdentifiersResult = itemRepositoryHelper.getItemIdentifiers(getContext(), offset, length,
                         parameters.getMetadataPrefix());
         } else {
-            if (!getRepository().getSetRepository().exists(parameters.getSet()) && !getContext().hasSet(parameters.getSet()))
-                throw new NoMatchesException();
+            /* TBH, I don't remember why the 2 lines below are commented out in 
+               the original IQSS fork. Will check. -- L.A. */
+            /*if (!getRepository().getSetRepository().exists(parameters.getSet()) && !getContext().hasSet(parameters.getSet()))
+                throw new NoMatchesException();*/
```

Comment:
I *think* there might have been a validation check that failed, looking at https://github.com/IQSS/dataverse/issues/3307.

The very same problem seems to be one of the causes for the existance of [XlistRecordsHandler.java](https://github.com/IQSS/dataverse/blob/50f093be5bf14eb940a0807e9c36c7e6b1806e4a/src/main/java/edu/harvard/iq/dataverse/harvest/server/xoai/XlistRecordsHandler.java#L86) (the rest of the code in there seems to be mostly the same as the original [ListRecordsHandler.java](https://github.com/gdcc/xoai/blob/branch-5.0/xoai-data-provider/src/main/java/io/gdcc/xoai/dataprovider/handlers/ListRecordsHandler.java)...)

I actually found a bug that might have caused the trouble leading to the workaround: 
https://github.com/DSpace/xoai/blob/c8857979d809938250b02f98087b2f14e5c24f51/xoai-data-provider/src/main/java/io/gdcc/xoai/dataprovider/handlers/helpers/SetRepositoryHelper.java#L64-L67

(Uses the wrong object for the comparison - should have been `setSpec`, **not** `set`).

I fixed this with https://github.com/gdcc/xoai/commit/149ac89f928df470793f62016d55ef4475cc28df and https://github.com/gdcc/xoai/commit/a95172d8472df75b96ad1b365116737e8415a822, implementing `Set.equals()` and using `List.contains()`, so I *think* we should be able to fix this one, too.